src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActor.scala

```diff
@@ -135,7 +142,25 @@ private[impl] class InstanceTrackerActor(
         val op = InstanceOpProcessor.Operation(deadline, sender(), instanceId, instanceUpdateOp)
         updaterRef.forward(InstanceUpdateActor.ProcessInstanceOp(op))
 
-      case msg @ InstanceTrackerActor.StateChanged(ack) =>
+      case InstanceTrackerActor.StateChanged(ack) =>
+        import context.dispatcher
+
+        ack.effect match {
+          case InstanceUpdateEffect.Update(instance, _, _) =>
+            updateRepository(() => repository.store(instance), ack)
+              .pipeTo(self)(sender())
+
+          case InstanceUpdateEffect.Expunge(instance, _) =>
+            logger.debug(s"Received expunge for ${instance.instanceId}")
+            updateRepository(() => repository.delete(instance.instanceId), ack)
+              .pipeTo(self)(sender())
```

Comment:
What is `pipeTo(self)(sender())` doing? Is the result sent to `self` and `sender()`?

Reply:
no the second parameter is the sender, so I wanted to make sure that when processing the second message, when calling `sender()` I will get sender of the first message... Maybe it's redundand and akka passes the original sender by default, I have to check... In any case I think it's not breaking anything, it's just redundant

Reply:
After consideration, do you think just doing onComplete and removing the updateRepository method might be more straightforward?

```
val originalSender = sender
repository.delete(instance.instanceId).onComplete {
  case Failure(NonFatal(ex)) =>
    originalSender ! Status.Failure(ex)
  case Success(_) =>
    self.tell(RepositoryStateUpdated(ack), originalSender)
}
```

Seems substantially shorter considering it would remove the private method, extra status and associated receive handler, etc.